libs/utils/include/utils/StructureOfArrays.h

```diff
@@ -166,7 +166,12 @@ class StructureOfArraysBase {
     template<typename CVQualifiedSOAPointer>
     class Iterator {
         friend class StructureOfArraysBase;
-        CVQualifiedSOAPointer UTILS_RESTRICT soa;
+        CVQualifiedSOAPointer 
+#ifndef _MSC_VER
+            // MSVC allows __restrict only for pointers and chokes here
```

Comment:
What's the exact error reported by MSVC here?

Reply:
```
1>C:\greg\github\filament\libs\utils\include\utils/StructureOfArrays.h(174,1): error C2219:  syntax error: type qualifier must be after '*'
```

Reply:
BTW, is there a good reason to use UTILS_RESTRICT here?

Reply:
Actually, looking into it more, it seems to me that using UTILS_RESTRICT is incorrect here. The meaning is that only this specific variable is used to access the pointed memory. Since this is a variable in an iterator that points to the `StructureOfArraysBase` data structure, and it is legal to create multiple iterators to that structure, thereby aliasing access to the SoA via two iterator::soa pointers.

Reply:
I think I should remove the UTILS_RESTRICT alltogether!

Reply:
yes. agreed.

Reply:
OK, I removed the UTILS_RESTRICT.